src/main.rs

```diff
@@ -228,12 +275,10 @@ async fn initialize_proxy(
         match monitor(router, abort_handles, epsilon, server_handle).await {
             Reconnect::NewUpstream(new_pool_addr) => {
```

Comment:
Use the `new_pool_addr` returned by monitor to update `pool_addr`  in single-upstream mode for the next loop iteration to reconnect to a better upstream pool, since monitoring is disabled for multi-upstream for now.
Introduction of multi upstream should not change how single upstream already works.